test/e2e/azure_test.go

```diff
@@ -91,6 +105,195 @@ var _ = Describe("Workoad cluster creation", func() {
+
+			By("Applying a network policy to allow egress access to app: webapp, role: frontend pods from pods with app: webapp, role: backend labels in same development namespace")
+			nwpolicyName, namespaceE2E, nwpolicyFileName = "backend-allow-egress-pod-namespace-label", nsDev, "backend-policy-allow-egress-pod-namespace-label.yaml"
+			e2e_networkpolicy.ApplyNetworkPolicy(nwpolicyName, namespaceE2E, nwpolicyFileName, PolicyDir)
+
+			By("Ensuring we have egress access from pods with matching labels")
+			e2e_networkpolicy.EnsureConnectivityResultBetweenPods(backendPods, frontendDevPods, validateNetworkPolicyTimeout, true)
+
+			By("Ensuring we don't have ingress access from pods without matching labels")
+			e2e_networkpolicy.EnsureConnectivityResultBetweenPods(backendPods, frontendProdPods, validateNetworkPolicyTimeout, false)
+			e2e_networkpolicy.EnsureConnectivityResultBetweenPods(backendPods, nwpolicyPods, validateNetworkPolicyTimeout, false)
+
+			By("Cleaning up after ourselves")
+			e2e_networkpolicy.DeleteNetworkPolicy(nwpolicyName, namespaceE2E)
+
+			By("Applying a network policy to only allow ingress access to app: webapp, role: backend pods in development namespace from pods in any namespace with the same labels")
+			nwpolicyName, namespaceE2E, nwpolicyFileName = "backend-allow-ingress-pod-label", nsDev, "backend-policy-allow-ingress-pod-label.yaml"
+			e2e_networkpolicy.ApplyNetworkPolicy(nwpolicyName, namespaceE2E, nwpolicyFileName, PolicyDir)
+
+			By("Ensuring we have ingress access from pods with matching labels")
+			e2e_networkpolicy.EnsureConnectivityResultBetweenPods(backendPods, backendPods, validateNetworkPolicyTimeout, true)
+
+			By("Ensuring we don't have ingress access from pods without matching labels")
+			e2e_networkpolicy.EnsureConnectivityResultBetweenPods(nwpolicyPods, backendPods, validateNetworkPolicyTimeout, false)
+
+			By("Cleaning up after ourselves")
+			e2e_networkpolicy.DeleteNetworkPolicy(nwpolicyName, namespaceE2E)
+
+			By("Applying a network policy to only allow ingress access to app: webapp role:backends in development namespace from pods with label app:webapp, role: frontendProd within namespace with label purpose: development")
+			nwpolicyName, namespaceE2E, nwpolicyFileName = "backend-policy-allow-ingress-pod-namespace-label", nsDev, "backend-policy-allow-ingress-pod-namespace-label.yaml"
+			e2e_networkpolicy.ApplyNetworkPolicy(nwpolicyName, namespaceE2E, nwpolicyFileName, PolicyDir)
+
+			By("Ensuring we don't have ingress access from role:frontend pods in production namespace")
+			e2e_networkpolicy.EnsureConnectivityResultBetweenPods(frontendProdPods, backendPods, validateNetworkPolicyTimeout, false)
+
+			By("Ensuring we have ingress access from role:frontend pods in development namespace")
+			e2e_networkpolicy.EnsureConnectivityResultBetweenPods(frontendDevPods, backendPods, validateNetworkPolicyTimeout, true)
+
+			By("Cleaning up after ourselves")
```

Comment:
Does it make sense to have an afterTest func to cleanup, or if the cluster is getting deleted anyway, do we even need to cleanup?

Reply:
yep, makes sense we dont need to cleanup. will remove

Reply:
@nader-ziada done, cleaned up some un-used functions